pkgs/development/python-modules/cut-cross-entropy/default.nix

```
@@ -0,0 +1,51 @@
+{ 
+  lib, buildPythonPackage, fetchFromGitHub, stdenv, setuptools, setuptools-scm, torch, triton,
+  transformers ? null,
+  deepspeed ? null,
+  accelerate ? null,
+  datasets ? null,
+  huggingface_hub ? null,
+  pandas ? null,
+  fire ? null,
+  tqdm ? null
+}:
+
+buildPythonPackage rec {
+  pname = "cut-cross-entropy";
+  version = "unstable-2024-03-10";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "apple";
+    repo = "ml-cross-entropy";
+    rev = "24fbe4b5dab9a6c250a014573613c1890190536c";
```

A small comment for futures editors
```suggestion
    rev = "24fbe4b5dab9a6c250a014573613c1890190536c"; # no tags
```